user/serializers.py

```diff
@@ -1 +1,27 @@
-# write your code here
+from django.contrib.auth import get_user_model
+from rest_framework import serializers
+
+
+class UserSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = get_user_model()
+        fields = ("id", "username", "password", "email", "is_staff")
+        read_only_fields = ("id", "is_staff")
+        extra_kwargs = {
+            "password": {"write_only": True, "min_length": 5},
+        }
+
+    def create(self, validated_data):
+        """create user with encrypted password"""
+        return get_user_model().objects.create_user(**validated_data)
+
+    def update(self, instance, validated_data):
+        """update user with encrypted password"""
+        password = validated_data.pop("password", None)
```

Comment:
This line is popping the password from the `validated_data` dictionary. If the password is not provided in the request, it will return `None`. This could potentially unset a user's password if an update request is made without a password. Consider checking if the password is in the `validated_data` before popping it.